File: CS/DataItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace DXGridThreads {
    public class DataItem : INotifyPropertyChanged {


        protected string _Name;

        public string Name {
            get {
                return this._Name;
            }

            set {
                if(this._Name != value) {
                    this._Name = value;
                    this.OnPropertyChanged("Name");
                }
            }
        }


        protected volatile int _Value;

        public int Value {
            get {
                return this._Value;
            }

            set {
                if(this._Value != value) {
                    this._Value = value;
                    this.OnPropertyChanged("Value");
                }
            }
        }


        public override string ToString() {
            return Name;
        }


        public void OnPropertyChanged(string info) {
            if(this.PropertyChanged != null) {
                this.PropertyChanged(this, new PropertyChangedEventArgs(info));
            }
        }

        public event PropertyChangedEventHandler PropertyChanged;
    }
}

File: VB/DataItem.vb
Imports System.ComponentModel

Namespace DXGridThreads
    Public Class DataItem
        Implements INotifyPropertyChanged


        Protected _Name As String

        Public Property Name() As String
            Get
                Return Me._Name
            End Get

            Set(ByVal value As String)
                If Me._Name <> value Then
                    Me._Name = value
                    Me.OnPropertyChanged("Name")
                End If
            End Set
        End Property

        Protected _Value As Integer

        Public Property Value() As Integer
            Get
                Return Me._Value
            End Get

            Set(ByVal value As Integer)
                If Me._Value <> value Then
                    Me._Value = value
                    Me.OnPropertyChanged("Value")
                End If
            End Set
        End Property


        Public Overrides Function ToString() As String
            Return Name
        End Function


        Public Sub OnPropertyChanged(ByVal info As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(info))
        End Sub

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace
